src/convertToObject.js

```diff
@@ -5,8 +5,106 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const newObject = {};
+
+  const splitStrings = sourceString.split(';');
+
+  const formateStrings = splitStrings.map((param) => {
+    const newString = param
+      .replaceAll('\n', '')
+      .replaceAll(',', `,\n`)
+      .split(':');
+
+    return newString;
+  });
+
+  formateStrings.forEach((element) => {
+    if (
+      typeof element[0] !== 'undefined' &&
+      typeof element[1] !== 'undefined'
+    ) {
+      const newKey = element[0].trim();
+      const newValue = element[1].trim();
+
+      return Object.assign(newObject, { [newKey]: `${newValue}` });
```

Comment:
The `return` statement inside a `forEach` loop is unnecessary since `forEach` does not utilize the return value. You can remove the `return` keyword here.